plaid/src/data/websocket/mod.rs

```diff
@@ -0,0 +1,408 @@
+mod selector;
+
+use crate::executor::Message;
+use crossbeam_channel::Sender;
+use futures_util::{
+    stream::{SplitSink, SplitStream},
+    SinkExt, StreamExt,
+};
+use http::{HeaderMap, Uri};
+use plaid_stl::messages::{Generator, LogSource, LogbacksAllowed};
+use selector::UriSelector;
+use serde::Deserialize;
+use std::{collections::HashMap, str::FromStr, time::Duration};
+use tokio::{net::TcpStream, task::JoinHandle};
+use tokio_tungstenite::{
+    connect_async,
+    tungstenite::{client::IntoClientRequest, protocol::Message as WSMessage},
+    MaybeTlsStream, WebSocketStream,
+};
+
+#[derive(Debug)]
+enum Errors {
+    SocketCreationFailure,
+}
+
+#[derive(Deserialize)]
+pub struct WebsocketDataGenerator {
+    websockets: HashMap<String, Websocket>,
+}
+
+#[derive(Deserialize)]
+pub struct Websocket {
+    /// The URI(s) of the WebSocket endpoint. The config allows multiple URIs in the event that one fails,
+    /// a new one can be used in its place.
+    #[serde(deserialize_with = "parse_uris")]
+    uris: Vec<Uri>,
+    /// A string indicating the type of log associated with the WebSocket.
+    log_type: String,
+    /// The message to be sent over the WebSocket connection.
```

Comment:
I think this needs more detail. Why are we sending messages over the connection?

Reply:
How many times, in what circumstance?